test/bin/local-solver-picks-higher-version.t/run.t

```diff
@@ -0,0 +1,57 @@
+With opam-0install solver, the context is responsible for defining the
+preference order between versions of a package.
+To stick to the default behaviour of the switch based context,
+the repository list based context should properly sort the candidates so
+that the solver picks the highest satisfying version.
+
+Here we define a package test that depends on a package `a`:
+
+  $ cat test.opam
+  opam-version: "2.0"
+  depends: [
+    "dune"
+    "a"
+  ]
+  x-opam-monorepo-opam-repositories: [
+    "file://$OPAM_MONOREPO_CWD/repo"
+  ]
+
+We have a local repo that defines two versions of `a`, both being satisfying
+solutions:
+
+  $ cat repo/packages/a/a.0.1/opam
+  opam-version: "2.0"
+  dev-repo: "git+https://a.com/a.git"
+  depends: [
+    "dune"
+  ]
+  url {
+    src: "https://a.com/a.0.1.tbz"
+    checksum: [
+      "sha256=0000000000000000000000000000000000000000000000000000000000000000"
+    ]
+  }
+  $ cat repo/packages/a/a.0.2/opam
+  opam-version: "2.0"
+  dev-repo: "git+https://a.com/a.git"
+  depends: [
+    "dune"
+  ]
+  url {
+    src: "https://a.com/a.0.2.tbz"
+    checksum: [
+      "sha256=0000000000000000000000000000000000000000000000000000000000000001"
+    ]
+  }
+
+opam-monorepo solver should pick a.0.2 here:
+
+  $ opam-monorepo lock
+  ==> Using 1 locally scanned package as the target.
+  ==> Found 9 opam dependencies for the target package.
+  ==> Querying opam database for their metadata and Dune compatibility.
+  ==> Calculating exact pins for each of them.
+  ==> Wrote lockfile with 1 entries to $TESTCASE_ROOT/test.opam.locked. You can now run opam monorepo pull to fetch their sources.
+  $ cat test.opam.locked | grep "\"a\""
```

Comment:
Maybe grep for `"a" {` to make "sure" that this is the `"a"` from the dependencies (or well, something like `"a"\s+\{`)

Reply:
I agree it would be nice to get rid of the other one from the `x-opam-monorepo-duniverse-dirs`. The only thing I was afraid of was that if we end up with `"a"` in the deps, it would show nothing, which is more confusing but now that I'm thinking about it, it's impossible to have that in the lockfile so I'll switch to your suggestion!